### ISSUE TITLE
Additional Documentation Needed

### ISSUE THREAD

**Issue by robert4man** [robert4man/cs3560_team1#1]
I have posted the code from HW5 into the repo with the initial tag. I assume we will utilize Doxygen. We should split up the work by file/function. We could create separate issues per file or group of functions to document then assign them to each member of the group so we do not duplicate effort.

**Comment by robert4man** [comment]
I have added an initial Doxyfile and empty doc dir for documentation output and tested that doxygen basically works as is. I have added a .gitignore to prevent generated documentation output from getting pushed back into the repo. Time to write some documentation.

**Comment by zhughes96** [comment]
When I log into putty, and open the cs3560_team1, I don't see anything except the readme. Where should I be looking?

**Comment by nathanbrinson** [comment]
I'll get working on othello.cc

**Comment by Odaslayer** [comment]
So I think Im kinda getting this assignment. Are we just supposed to edit the source code with these code blocks /*! so that Doxygen will be able to tell what comments there are?

**Comment by nathanbrinson** [comment]
All he wants is one function for othello and main but the whole thing for game. I'll work on play display_message and winning in game.

**Comment by nathanbrinson** [comment]
and get user move too lol

**Comment by robert4man** [comment]
I'll be on-line around 9:30 PM to help with this. I'll update the Makefile to include a doc target and fill in some doc blocks where ever needed.

**Comment by zhughes96** [comment]
I'm studying but I'll be able to work on it in a bit. Lmk if you guys need something!

**Comment by robert4man** [comment]
@nathanbrinson: why did you commit the html and latex dirs to the git repo? I crafted the Doxyfile so they would be created in the doc dir and ignored by .gitignore. It's not wise to commit dynamically generated content to the repo.

**Comment by robert4man** [comment]
I've updated the Makefile is a doc target and extended clean to clean up the dynamically generated doc files from under doc.

**Comment by robert4man** [comment]
I have removed the dynamically generated doc files to avoid losing points on the assignment.

**Comment by Odaslayer** [comment]
ok so when we document the source code for doxygen, is the source code the only thing we are pushing to github?

**Comment by Odaslayer** [comment]
I added a bit of documentation to the colors.h file and didnt run Doxygen so it wouldnt make another html and latex directory. Please let me know if I am doing anything wrong!

**Comment by robert4man** [comment]
Yes, the updated source code is all you should be pushing into github. I have added a doc target to the Makefile so you should be able to use `make doc` to run doxygen and it should generate the documentation under the doc directory. There is a.gitignore in this location to prevent anything created under doc from getting added to the repo, but you can also run `make clean` to remove it if you desire. I have provided a more or less complete piece.h.

**Comment by Odaslayer** [comment]
Working on game.h now

**Comment by Odaslayer** [comment]
Just completed documenting and pushing game.h

**Comment by robert4man** [comment]
It's not exactly complete or perfect, but I'm going to go ahead and tag it v2.DocReady so we can complete the submissions.

**Comment by Odaslayer** [comment]
ok sounds good. For the generated documentation part of the submission, does he just want our html file zipped up?

**Comment by zhughes96** [comment]
Oh, hey guys I can work on the project, is there anything left to do? Do we all have to submit the same things?

**Comment by zhughes96** [comment]
I'll do make_human_move in game.cc

**Comment by zhughes96** [comment]
Did anyone else get a long list of code for when they ran make all?

**Comment by robert4man** [comment]
Most of the output from `make all` is generated by the output from doxygen. I have added QUIET=yes and WARN_IF_UNDOCUMENTED=no in the Doxyfile to make it a little more quiet. However, WARN_IF_UNDOCUMENTED can be useful for determining items which still need documented.
